src/components/mortgage-investment-compare/index.js

```diff
@@ -49,17 +61,41 @@ class MortgageInvestmentCompare extends React.Component {
 
     const result = this.getResult(input)
 
-    this.state = {
+    this.setState({
       input,
-      result,
-      showTable: false
-    }
+      result
+    })
   }
 
   toggleShowTable () {
     this.setState({ showTable: !this.state.showTable })
   }
 
+  toggleIncludeROI () {
+    this.setState({ includeROI: !this.state.includeROI }, () => {
+      const input = _.cloneDeep(this.state.input)
+      input.investmentRate = this.state.includeROI
+        ? this.state.input.investmentRate
+        : 0
+
+      const result = this.getResult(input)
+      this.setState({ result })
+    })
+  }
```

Comment:
fyi, `setState` takes a callback as a second argument which is called after setting the state. So I'm setting the state and changing the boolean for `includeROI`, and then after that is done, running `getResult` and setting state again. I don't particularly love this implementation but it works.

Reply:
I'm going to change this a little. I'll be using lifecycle methods instead of callbacks. Will push my changes within the next hour

Reply:
took care of it here: https://github.com/jgozal/moneycompass/pull/61/commits/052412c46b114690bf0fe7a5c315d77adaf65a81

Reply:
What was happening before you made the optimization?

Reply:
@hdavidzhu I was calling `getResult` and setting the state for a second time as a callback to `setState`. Not really the best way to do it